resharper/resharper-unity/src/CSharp/Feature/Services/QuickFixes/MoveQuickFixes/MonoBehaviourMoveUtil.cs

```diff
@@ -0,0 +1,263 @@
+
+            RenameOldUsages(expression, result, name, factory);
+        }
+
+        public static void RenameOldUsages([NotNull]ICSharpExpression originExpression, [CanBeNull]IDeclaredElement localVariableDeclaredElement, 
+            [NotNull] string newName, [NotNull] CSharpElementFactory factory)
+        {
+            originExpression.GetPsiServices().Locks.AssertReadAccessAllowed();
+            originExpression.GetPsiServices().Locks.AssertMainThread();
+
+            var statement = ExpressionStatementNavigator.GetByExpression(originExpression);
+            if (statement != null)
+            {
+                statement.RemoveOrReplaceByEmptyStatement();
+            }
+            else
+            {
+                if (localVariableDeclaredElement == null)
+                {
+                    originExpression.ReplaceBy(factory.CreateReferenceExpression(newName));
+                }
+                else if (!newName.Equals(localVariableDeclaredElement.ShortName))
+                {
+                    var provider = DefaultUsagesProvider.Instance;
+                    var usages = provider.GetUsages(localVariableDeclaredElement,
+                        originExpression.GetContainingNode<IMethodDeclaration>().NotNull("scope != null"));
+                    originExpression.GetContainingStatement().NotNull("expression.GetContainingStatement() != null")
+                        .RemoveOrReplaceByEmptyStatement();
+                    foreach (var usage in usages)
+                    {
+                        if (usage.IsValid() && usage is IReferenceExpression node)
+                            node.ReplaceBy(factory.CreateReferenceExpression(newName));
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// If current expression is used as initializer for local variable, declared element for this variable will be returned
+        /// </summary>
+        public static IDeclaredElement GetDeclaredElementFromParentDeclaration([NotNull] ICSharpExpression expression)
+        {
+            expression.GetPsiServices().Locks.AssertReadAccessAllowed();
+            
+            var localVariableDeclaration =
+                LocalVariableDeclarationNavigator.GetByInitial(
+                    expression.GetContainingParenthesizedExpression()?.Parent as IVariableInitializer);
+            return localVariableDeclaration?.DeclaredElement;
+        }
+
+        public static string CreateBaseName([NotNull]ICSharpExpression toMove, [CanBeNull] IDeclaredElement variableDeclaration)
+        {
+            toMove.GetPsiServices().Locks.AssertReadAccessAllowed();
+            
+            var type = toMove.Type(new ResolveContext(toMove.GetPsiModule()));
+            if (type.IsUnknown)
+                type = TypeFactory.CreateTypeByCLRName("System.Object", toMove.GetPsiModule());
+            
+            string baseName =  type.GetPresentableName(CSharpLanguage.Instance);
```

Comment:
`.GetPresentableName()` returns a string which should represent a human-readable type name in hints/warnings. It might not be a valid identifier e.g. it could return `int[]` or `MyType<T>`.
Does naming service handle such `base names`? Please, add tests for such scenarios.